src/shared/tar-util.c

```diff
@@ -1138,31 +1166,37 @@ static int archive_write_acl(
 
                         qualifier = *q;
                         sym_acl_free(q);
+
+                        /* Suppress invalid UIDS or those that shall be squashed */
+                        skip = uid_is_valid(qualifier) &&
+                                (!FLAGS_SET(flags, TAR_SQUASH_UIDS_ABOVE_64K) || qualifier <= 0xFFFFU);
```

Comment:
Please add parentheses around the `&&`. Also I don't get the condition here, you're skipping valid uids if the squash flag isn't set or they're within some range? Shouldn't it be the opposite?